src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -2628,12 +2761,12 @@ public DeferredPermissionsSource(TaskListener listener) {
         public GHPermissionType fetch(String username) throws IOException, InterruptedException {
             if (repo == null) {
                 listener.getLogger().format("Connecting to %s to check permissions of obtain list of %s for %s/%s%n",
-                        apiUri == null ? GITHUB_URL : apiUri, username, repoOwner, repository);
+                        apiUri, username, repoOwner, repository);
                 StandardCredentials credentials = Connector.lookupScanCredentials(
                         (Item) getOwner(), apiUri, credentialsId
                 );
                 github = Connector.connect(apiUri, credentials);
-                String fullName = repoOwner + "/" + repository;
+                String fullName = repoOwner + "/" + removeEnd(repository, ".git");
```

Comment:
Ditto, I think this would be cause issues in other places as well, this is just one of the only places the tests catch it. I think it should be done in `GitHubRepositoryInfo`.